### PR TITLE
Fix view rights checks on `Config` class

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #15302

### CODE REVIEW

File: src/Config.php [hunk 1]
```diff
@@ -118,9 +118,11 @@ public static function canCreate()
...
     {
         if (
-            isset($this->fields['context']) &&
-            ($this->fields['context'] == 'core' ||
-            Plugin::isPluginActive($this->fields['context']))
+            isset($this->fields['context'])
+            && (
+                in_array($this->fields['context'], ['core', 'inventory'], true) // GLPI config contexts
+                || Plugin::isPluginActive($this->fields['context'])
+            )
```
Comment (jon-nfc) [inline src/Config.php:125]: @cedric-anne, thanks for the speedy reply.

I've tested this change and it only makes it visible, an api call to `http://127.0.0.1/apirest.php/Config?range=0-1000` still doesn't incude the inventory settings.

also editing still gives permission denied

Comment (jon-nfc) [reply src/Config.php:125]: correction to my last, i can edit the setting (opps, needed HTTP/PUT), however fetching all settings with url `http://127.0.0.1/apirest.php/Config?range=0-1000` doesn't return inventory items

Comment (cedric-anne) [reply src/Config.php:125]: I added a commit that should fix this.

File: src/Search.php [hunk 2]
```diff
@@ -4464,7 +4464,7 @@ public static function addDefaultWhere($itemtype)
                 break;
 
             case 'Config':
-                $availableContexts = ['core'] + Plugin::getPlugins();
+                $availableContexts = array_merge(['core', 'inventory'], Plugin::getPlugins());
```
Comment (jon-nfc) [inline src/Search.php:4467]: Legend, tested and works as intended. 

I can confirm this will resolve my bug report

Comment (cedric-anne) [reply src/Search.php:4467]: :+1:

**Review by trasher — APPROVED** [review]

**Pull request merged**